src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -176,6 +177,14 @@ public function set($id, $service)
         if (null === $service) {
             unset($this->services[$id]);
         }
+
+        if (isset($this->privates[$id])) {
+            if (null === $service) {
+                @trigger_error(sprintf('Unsetting the "%s" private service from outside of the container is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0. An exception will be thrown instead in Symfony 4.0.', $id), E_USER_DEPRECATED);
+            } else {
+                @trigger_error(sprintf('Resetting the "%s" private service from outside of the container is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0. An exception will be thrown instead in Symfony 4.0.', $id), E_USER_DEPRECATED);
+            }
```

Comment:
I agree with @fabpot here, it's the same for private properties on objects btw: they exists from the inside but not from the outside so that one can use the prop name without any collision.